# Import Libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# About Dataset

* EmployeeID: Unique identifier for each employee.
* FirstName: The first name of the employee.
* LastName: The last name of the employee.
* Gender: The gender of the employee.
* Age: The age of the employee.
* BusinessTravel: The frequency of business travel for the employee.
* Department: The department in which the employee works.
* DistanceFromHome (KM): The distance between the employee's home and workplace in kilometers.
* State: The state in which the employee resides.
* Ethnicity: The ethnicity of the employee.
* MaritalStatus: The marital status of the employee.
* Salary: The annual salary of the employee.
* StockOptionLevel: The level of stock options granted to the employee.
* OverTime: Whether the employee works overtime (Yes/No).
* HireDate: The date the employee was hired.
* Attrition: Whether the employee has left the company (Yes/No).
* YearsAtCompany: The number of years the employee has been with the company.
* YearsInMostRecentRole: The number of years the employee has been in their most recent role.
* YearsSinceLastPromotion: The number of years since the employee's last promotion.
* YearsWithCurrManager: The number of years the employee has worked with their current manager.
* EducationLevelID: Unique identifier for the education level.
* EducationLevel: The level of education achieved, ranging from "No Formal Qualifications" to "Doctorate."


In [4]:
data=pd.read_csv('Employee.csv')

In [5]:
data

EmployeeID   FirstName     LastName      Gender  Age BusinessTravel  \
0     3012-1A41    Leonelle        Simco      Female   30    Some Travel   
1     CBCB-9C9D     Leonerd        Aland        Male   38    Some Travel   
2     95D7-1CE9       Ahmed        Sykes        Male   43    Some Travel   
3     47A0-559B  Ermentrude       Berrie  Non-Binary   39    Some Travel   
4     42CC-040A       Stace       Savege      Female   29    Some Travel   
...         ...         ...          ...         ...  ...            ...   
1465  467E-977A         Jud    Melanaphy        Male   20    Some Travel   
1466  6FB9-A624        Marc       Calver  Non-Binary   27    Some Travel   
1467  EBF4-5928     Rudolph  MacDearmont        Male   21    Some Travel   
1468  60E6-B1D9      Merill          Agg        Male   21    Some Travel   
1469  84D4-D4C3       Naoma      Hebbard      Female   20     No Travel    

           Department  DistanceFromHome (KM) State                  Ethnicity  \
0               Sales                     27    IL                      White   
1               Sales                     23    CA                      White   
2     Human Resources                     29    CA    Asian or Asian American   
3          Technology                     12    IL                      White   
4     Human Resources                     29    CA                      White   
...               ...                    ...   ...                        ...   
1465       Technology                     28    CA  Black or African American   
1466       Technology                      8    CA  Black or African American   
1467            Sales                      4    CA  Black or African American   
1468       Technology                      7    CA  Black or African American   
1469       Technology                     28    CA  Black or African American   

      ...  MaritalStatus  Salary StockOptionLevel OverTime    HireDate  \
0     ...       Divorced  102059                1       No  2012-01-03   
1     ...         Single  157718                0      Yes  2012-01-04   
2     ...        Married  309964                1       No  2012-01-04   
3     ...        Married  293132                0       No  2012-01-05   
4     ...         Single   49606                0       No  2012-01-05   
...   ...            ...     ...              ...      ...         ...   
1465  ...         Single   38508                0       No  2022-12-26   
1466  ...         Single   92995                3       No  2022-12-30   
1467  ...         Single   63375                0       No  2022-12-30   
1468  ...        Married   46521                0       No  2022-12-30   
1469  ...        Married   38258                0       No  2022-12-31   

      Attrition YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
0            No             10                     4                       9   
1            No             10                     6                      10   
2            No             10                     6                      10   
3            No             10                    10                      10   
4           Yes              6                     1                       1   
...         ...            ...                   ...                     ...   
1465         No              0                     0                       0   
1466         No              0                     0                       0   
1467         No              0                     0                       0   
1468         No              0                     0                       0   
1469         No              0                     0                       0   

      YearsWithCurrManager  
0                        7  
1                        0  
2                        8  
3                        0  
4                        6  
...                    ...  
1465                     0  
1466                     0  
1467    

In [6]:
data.columns

Index(['EmployeeID', 'FirstName', 'LastName', 'Gender', 'Age',
       'BusinessTravel', 'Department', 'DistanceFromHome (KM)', 'State',
       'Ethnicity', 'Education', 'EducationField', 'JobRole', 'MaritalStatus',
       'Salary', 'StockOptionLevel', 'OverTime', 'HireDate', 'Attrition',
       'YearsAtCompany', 'YearsInMostRecentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')

In [8]:
data.head()

EmployeeID   FirstName LastName      Gender  Age BusinessTravel  \
0  3012-1A41    Leonelle    Simco      Female   30    Some Travel   
1  CBCB-9C9D     Leonerd    Aland        Male   38    Some Travel   
2  95D7-1CE9       Ahmed    Sykes        Male   43    Some Travel   
3  47A0-559B  Ermentrude   Berrie  Non-Binary   39    Some Travel   
4  42CC-040A       Stace   Savege      Female   29    Some Travel   

        Department  DistanceFromHome (KM) State                Ethnicity  ...  \
0            Sales                     27    IL                    White  ...   
1            Sales                     23    CA                    White  ...   
2  Human Resources                     29    CA  Asian or Asian American  ...   
3       Technology                     12    IL                    White  ...   
4  Human Resources                     29    CA                    White  ...   

   MaritalStatus  Salary StockOptionLevel OverTime    HireDate  Attrition  \
0       Divorced  102059                1       No  2012-01-03         No   
1         Single  157718                0      Yes  2012-01-04         No   
2        Married  309964                1       No  2012-01-04         No   
3        Married  293132                0       No  2012-01-05         No   
4         Single   49606                0       No  2012-01-05        Yes   

  YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
0             10                     4                       9   
1             10                     6                      10   
2             10                     6                      10   
3             10                    10                      10   
4              6                     1                       1   

   YearsWithCurrManager  
0                     7  
1                     0  
2                     8  
3                     0  
4                     6  

[5 rows x 23 columns]

In [9]:
data.tail()

EmployeeID FirstName     LastName      Gender  Age BusinessTravel  \
1465  467E-977A       Jud    Melanaphy        Male   20    Some Travel   
1466  6FB9-A624      Marc       Calver  Non-Binary   27    Some Travel   
1467  EBF4-5928   Rudolph  MacDearmont        Male   21    Some Travel   
1468  60E6-B1D9    Merill          Agg        Male   21    Some Travel   
1469  84D4-D4C3     Naoma      Hebbard      Female   20     No Travel    

      Department  DistanceFromHome (KM) State                  Ethnicity  ...  \
1465  Technology                     28    CA  Black or African American  ...   
1466  Technology                      8    CA  Black or African American  ...   
1467       Sales                      4    CA  Black or African American  ...   
1468  Technology                      7    CA  Black or African American  ...   
1469  Technology                     28    CA  Black or African American  ...   

      MaritalStatus Salary StockOptionLevel OverTime    HireDate  Attrition  \
1465         Single  38508                0       No  2022-12-26         No   
1466         Single  92995                3       No  2022-12-30         No   
1467         Single  63375                0       No  2022-12-30         No   
1468        Married  46521                0       No  2022-12-30         No   
1469        Married  38258                0       No  2022-12-31         No   

     YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
1465              0                     0                       0   
1466              0                     0                       0   
1467              0                     0                       0   
1468              0                     0                       0   
1469              0                     0                       0   

      YearsWithCurrManager  
1465                     0  
1466                     0  
1467                     0  
1468                     0  
1469                     0  

[5 rows x 23 columns]

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   EmployeeID               1470 non-null   object
 1   FirstName                1470 non-null   object
 2   LastName                 1470 non-null   object
 3   Gender                   1470 non-null   object
 4   Age                      1470 non-null   int64 
 5   BusinessTravel           1470 non-null   object
 6   Department               1470 non-null   object
 7   DistanceFromHome (KM)    1470 non-null   int64 
 8   State                    1470 non-null   object
 9   Ethnicity                1470 non-null   object
 10  Education                1470 non-null   int64 
 11  EducationField           1470 non-null   object
 12  JobRole                  1470 non-null   object
 13  MaritalStatus            1470 non-null   object
 14  Salary                   1470 non-null  

# Key points
* Data Completeness: All columns have 1,470 non-null entries, indicating no missing data, which simplifies analysis.
* Data Types:
    * Categorical Variables: Gender, BusinessTravel, Department, State, Ethnicity, EducationField, JobRole, MaritalStatus, OverTime, Attrition.
    * Numerical Variables: Age, DistanceFromHome, Education, Salary, StockOptionLevel, YearsAtCompany, YearsInMostRecentRole, YearsSinceLastPromotion, YearsWithCurrManager.
    * Date Variable: HireDate.

In [11]:
data.describe()

Age  DistanceFromHome (KM)    Education         Salary  \
count  1470.000000            1470.000000  1470.000000    1470.000000   
mean     28.989796              22.502721     2.912925  112956.497959   
std       7.993055              12.811124     1.024165  103342.889222   
min      18.000000               1.000000     1.000000   20387.000000   
25%      23.000000              12.000000     2.000000   43580.500000   
50%      26.000000              22.000000     3.000000   71199.500000   
75%      34.000000              33.000000     4.000000  142055.750000   
max      51.000000              45.000000     5.000000  547204.000000   

       StockOptionLevel  YearsAtCompany  YearsInMostRecentRole  \
count       1470.000000     1470.000000            1470.000000   
mean           0.793878        4.562585               2.293197   
std            0.852077        3.288048               2.539093   
min            0.000000        0.000000               0.000000   
25%            0.000000        2.000000               0.000000   
50%            1.000000        4.000000               1.000000   
75%            1.000000        7.000000               4.000000   
max            3.000000       10.000000              10.000000   

       YearsSinceLastPromotion  YearsWithCurrManager  
count              1470.000000           1470.000000  
mean                  3.440816              2.239456  
std                   2.945194              2.505774  
min                   0.000000              0.000000  
25%                   1.000000              0.000000  
50%                   3.000000              1.000000  
75%                   6.000000              4.000000  
max                  10.000000             10.000000

In [12]:
data.describe(include='all')

EmployeeID FirstName LastName  Gender          Age BusinessTravel  \
count        1470      1470     1470    1470  1470.000000           1470   
unique       1470      1334     1441       4          NaN              3   
top     3012-1A41   Murdock   Ponten  Female          NaN    Some Travel   
freq            1         3        2     675          NaN           1043   
mean          NaN       NaN      NaN     NaN    28.989796            NaN   
std           NaN       NaN      NaN     NaN     7.993055            NaN   
min           NaN       NaN      NaN     NaN    18.000000            NaN   
25%           NaN       NaN      NaN     NaN    23.000000            NaN   
50%           NaN       NaN      NaN     NaN    26.000000            NaN   
75%           NaN       NaN      NaN     NaN    34.000000            NaN   
max           NaN       NaN      NaN     NaN    51.000000            NaN   

        Department  DistanceFromHome (KM) State Ethnicity  ...  MaritalStatus  \
count         1470            1470.000000  1470      1470  ...           1470   
unique           3                    NaN     3         7  ...              3   
top     Technology                    NaN    CA     White  ...        Married   
freq           961                    NaN   875       860  ...            624   
mean           NaN              22.502721   NaN       NaN  ...            NaN   
std            NaN              12.811124   NaN       NaN  ...            NaN   
min            NaN               1.000000   NaN       NaN  ...            NaN   
25%            NaN              12.000000   NaN       NaN  ...            NaN   
50%            NaN              22.000000   NaN       NaN  ...            NaN   
75%            NaN              33.000000   NaN       NaN  ...            NaN   
max            NaN              45.000000   NaN       NaN  ...            NaN   

               Salary StockOptionLevel OverTime    HireDate  Attrition  \
count     1470.000000      1470.000000     1470        1470       1470   
unique            NaN              NaN        2        1192          2   
top               NaN              NaN       No  2016-10-04         No   
freq              NaN              NaN     1054           4       1233   
mean    112956.497959         0.793878      NaN         NaN        NaN   
std     103342.889222         0.852077      NaN         NaN        NaN   
min      20387.000000         0.000000      NaN         NaN        NaN   
25%      43580.500000         0.000000      NaN         NaN        NaN   
50%      71199.500000         1.000000      NaN         NaN        NaN   
75%     142055.750000         1.000000      NaN         NaN        NaN   
max     547204.000000         3.000000      NaN         NaN        NaN   

       YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
count     1470.000000           1470.000000             1470.000000   
unique            NaN                   NaN                     NaN   
top               NaN                   NaN                     NaN   
freq              NaN                   NaN                     NaN   
mean         4.562585              2.293197                3.440816   
std          3.288048              2.539093                2.945194   
min          0.000000              0.000000                0.000000   
25%          2.000000              0.000000                1.000000   
50%          4.000000              1.000000                3.000000   
75%          7.000000              4.000000                6.000000   
max         10.000000             10.000000               10.000000   

        YearsWithCurrManager  
count            1470.000000  
unique                   NaN  
top                      NaN  
freq                     NaN  
mean                2.239456  
std                 2.505774  
min                 0.000000  
25%                 0.000000  
50%                 1.000000  
75%                 4.000000  
max                10.000000  

[11 ro

In [13]:
data.describe(include='object')

EmployeeID FirstName LastName  Gender BusinessTravel  Department State  \
count        1470      1470     1470    1470           1470        1470  1470   
unique       1470      1334     1441       4              3           3     3   
top     3012-1A41   Murdock   Ponten  Female    Some Travel  Technology    CA   
freq            1         3        2     675           1043         961   875   

       Ethnicity    EducationField          JobRole MaritalStatus OverTime  \
count       1470              1470             1470          1470     1470   
unique         7                 9               13             3        2   
top        White  Computer Science  Sales Executive       Married       No   
freq         860               440              327           624     1054   

          HireDate Attrition  
count         1470      1470  
unique        1192         2  
top     2016-10-04        No  
freq             4      1233

# Checking null values

In [14]:
data.isnull().sum()

EmployeeID                 0
FirstName                  0
LastName                   0
Gender                     0
Age                        0
BusinessTravel             0
Department                 0
DistanceFromHome (KM)      0
State                      0
Ethnicity                  0
Education                  0
EducationField             0
JobRole                    0
MaritalStatus              0
Salary                     0
StockOptionLevel           0
OverTime                   0
HireDate                   0
Attrition                  0
YearsAtCompany             0
YearsInMostRecentRole      0
YearsSinceLastPromotion    0
YearsWithCurrManager       0
dtype: int64

* No null values

# Checking duplicates in data

In [16]:
data.duplicated().sum()

0

* No duplicate values

# Converting datatype

In [18]:
data['HireDate']=data['HireDate'].astype('datetime64[ns]')

In [19]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   EmployeeID               1470 non-null   object        
 1   FirstName                1470 non-null   object        
 2   LastName                 1470 non-null   object        
 3   Gender                   1470 non-null   object        
 4   Age                      1470 non-null   int64         
 5   BusinessTravel           1470 non-null   object        
 6   Department               1470 non-null   object        
 7   DistanceFromHome (KM)    1470 non-null   int64         
 8   State                    1470 non-null   object        
 9   Ethnicity                1470 non-null   object        
 10  Education                1470 non-null   int64         
 11  EducationField           1470 non-null   object        
 12  JobRole                  1470 non-

In [20]:
data.head()

EmployeeID   FirstName LastName      Gender  Age BusinessTravel  \
0  3012-1A41    Leonelle    Simco      Female   30    Some Travel   
1  CBCB-9C9D     Leonerd    Aland        Male   38    Some Travel   
2  95D7-1CE9       Ahmed    Sykes        Male   43    Some Travel   
3  47A0-559B  Ermentrude   Berrie  Non-Binary   39    Some Travel   
4  42CC-040A       Stace   Savege      Female   29    Some Travel   

        Department  DistanceFromHome (KM) State                Ethnicity  ...  \
0            Sales                     27    IL                    White  ...   
1            Sales                     23    CA                    White  ...   
2  Human Resources                     29    CA  Asian or Asian American  ...   
3       Technology                     12    IL                    White  ...   
4  Human Resources                     29    CA                    White  ...   

   MaritalStatus  Salary StockOptionLevel OverTime   HireDate  Attrition  \
0       Divorced  102059                1       No 2012-01-03         No   
1         Single  157718                0      Yes 2012-01-04         No   
2        Married  309964                1       No 2012-01-04         No   
3        Married  293132                0       No 2012-01-05         No   
4         Single   49606                0       No 2012-01-05        Yes   

  YearsAtCompany YearsInMostRecentRole YearsSinceLastPromotion  \
0             10                     4                       9   
1             10                     6                      10   
2             10                     6                      10   
3             10                    10                      10   
4              6                     1                       1   

   YearsWithCurrManager  
0                     7  
1                     0  
2                     8  
3                     0  
4                     6  

[5 rows x 23 columns]

# Calculate the percentage of employees who left the company.

In [23]:
attrition_rate = data['Attrition'].value_counts(normalize=True) * 100
attrition_rate

No     83.877551
Yes    16.122449
Name: Attrition, dtype: float64

# Analyze if factors like age affect attrition rates

In [25]:
data.groupby(['Attrition'])['Age'].mean()

Attrition
No     29.460665
Yes    26.540084
Name: Age, dtype: float64

# Analyze if factors like department affect attrition rates

In [26]:
data.groupby(['Attrition'])['Department'].value_counts(normalize=True)

Attrition  Department     
No         Technology         0.671533
           Sales              0.287105
           Human Resources    0.041363
Yes        Technology         0.561181
           Sales              0.388186
           Human Resources    0.050633
Name: Department, dtype: float64

# Analyze if factors like job role affect attrition rates

In [27]:
data.groupby(['Attrition'])['JobRole'].value_counts(normalize=True)

Attrition  JobRole                  
No         Sales Executive              0.218978
           Software Engineer            0.200324
           Data Scientist               0.161395
           Machine Learning Engineer    0.110300
           Senior Software Engineer     0.099757
           Engineering Manager          0.059205
           Sales Representative         0.040552
           Analytics Manager            0.039740
           Manager                      0.028386
           HR Executive                 0.020276
           Recruiter                    0.012165
           HR Business Partner          0.005677
           HR Manager                   0.003244
Yes        Data Scientist               0.261603
           Sales Executive              0.240506
           Software Engineer            0.198312
           Sales Representative         0.139241
           Machine Learning Engineer    0.042194
           Recruiter                    0.037975
           Senior Software Engin

# Analyze if factors like salary affect attrition rates

In [28]:
data.groupby(['Attrition'])['Salary'].value_counts(normalize=True)

Attrition  Salary
No         26973     0.001622
           44082     0.001622
           44646     0.001622
           62828     0.001622
           65642     0.001622
                       ...   
Yes        341971    0.004219
           350525    0.004219
           377203    0.004219
           531629    0.004219
           539998    0.004219
Name: Salary, Length: 1460, dtype: float64

# Analyze if factors like overtime affect attrition rates

In [29]:
data.groupby(['Attrition'])['OverTime'].value_counts(normalize=True)

Attrition  OverTime
No         No          0.765612
           Yes         0.234388
Yes        Yes         0.535865
           No          0.464135
Name: OverTime, dtype: float64

# which departments have the highest or lowest average salaries.

In [33]:
data.groupby(['Department'])['Salary'].mean()

Department
Human Resources    119698.809524
Sales              119117.609865
Technology         109655.122789
Name: Salary, dtype: float64

# Salary range within different job roles.

In [34]:
data.groupby('JobRole')['Salary'].describe()

count           mean            std       min  \
JobRole                                                                    
Analytics Manager           52.0  346484.230769  118080.676209  107863.0   
Data Scientist             261.0   56079.494253   43547.070678   20387.0   
Engineering Manager         75.0  286258.506667   91001.990610  115110.0   
HR Business Partner          7.0  314002.428571  104142.207510  190466.0   
HR Executive                28.0   94362.321429   43767.654180   41745.0   
HR Manager                   4.0  449330.750000   92157.549484  315071.0   
Machine Learning Engineer  146.0  130164.616438   76486.269014   39456.0   
Manager                     37.0  317531.054054   95375.631038  135407.0   
Recruiter                   24.0   37647.500000   15103.645094   20583.0   
Sales Executive            327.0  117195.538226   66723.920446   37035.0   
Sales Representative        83.0   40656.421687   16882.717082   20650.0   
Senior Software Engineer   132.0  126161.295455   72674.852989   36012.0   
Software Engineer          294.0   51967.051020   37843.241005   20418.0   

                                 25%       50%        75%       max  
JobRole                                                              
Analytics Manager          252839.50  355858.5  448177.75  547204.0  
Data Scientist              32456.00   44057.0   65416.00  455643.0  
Engineering Manager        215367.50  292500.0  336153.50  539998.0  
HR Business Partner        229615.00  309964.0  396225.50  445906.0  
HR Executive                55655.25   89376.0  131310.75  201564.0  
HR Manager                 426056.00  486156.0  509430.75  509940.0  
Machine Learning Engineer   69932.50   98598.5  194454.00  366078.0  
Manager                    265712.00  313891.0  377203.00  513262.0  
Recruiter                   26099.00   30853.5   48956.50   70703.0  
Sales Executive             65313.00  100314.0  149126.00  350525.0  
Sales Representative        27122.50   36101.0   48534.50   97794.0  
Senior Software Engineer    68757.25  106898.5  167931.00  380248.0  
Software Engineer           30749.50   42704.0   60771.75  439641.0

# Gender balance in the organization.

In [35]:
data['Gender'].value_counts(normalize=True) * 100

Female               45.918367
Male                 44.285714
Non-Binary            8.435374
Prefer Not To Say     1.360544
Name: Gender, dtype: float64

# Analyze salary differences between male and female employees.

In [36]:
data.groupby('Gender')['Salary'].mean()

Gender
Female               113935.691852
Male                 111773.809524
Non-Binary           111395.508065
Prefer Not To Say    128083.350000
Name: Salary, dtype: float64

# How many employees work overtime and which departments or job roles have more overtime.

In [37]:
data['OverTime'].value_counts()

No     1054
Yes     416
Name: OverTime, dtype: int64

In [38]:
data.groupby('Department')['OverTime'].value_counts(normalize=True)

Department       OverTime
Human Resources  No          0.730159
                 Yes         0.269841
Sales            No          0.713004
                 Yes         0.286996
Technology       No          0.718002
                 Yes         0.281998
Name: OverTime, dtype: float64

# Working overtime affects the likelihood of employees leaving the company.

In [39]:
data.groupby('OverTime')['Attrition'].value_counts(normalize=True)

OverTime  Attrition
No        No           0.895636
          Yes          0.104364
Yes       No           0.694712
          Yes          0.305288
Name: Attrition, dtype: float64

# How long employees typically stay at the company.

In [41]:
data['YearsAtCompany'].mean()

4.562585034013606

# How often employees are promoted and if there are differences between departments.

In [42]:
data.groupby('Department')['YearsSinceLastPromotion'].mean()

Department
Human Resources    3.285714
Sales              3.260090
Technology         3.534860
Name: YearsSinceLastPromotion, dtype: float64

# Relationship between the education field and the type of job roles employees hold.

In [43]:
data.groupby('EducationField')['JobRole'].value_counts()

EducationField       JobRole                  
Business Studies     Software Engineer             39
                     Data Scientist                19
                     Machine Learning Engineer     14
                     Senior Software Engineer      14
                     Engineering Manager            5
                     Analytics Manager              3
Computer Science     Software Engineer            133
                     Data Scientist               121
                     Machine Learning Engineer     71
                     Senior Software Engineer      61
                     Engineering Manager           34
                     Analytics Manager             20
Economics            Sales Executive               65
                     Sales Representative          18
                     Recruiter                      6
                     HR Executive                   5
                     Manager                        5
                     HR Business Pa

#  How different education levels affect salary.

In [44]:
data.groupby('Education')['Salary'].mean()

Education
1     94983.482353
2    105180.535461
3    115405.430070
4    117641.057789
5    154268.791667
Name: Salary, dtype: float64

# Employees who live farther away are more likely to leave.

In [45]:
data.groupby('Attrition')['DistanceFromHome (KM)'].mean()

Attrition
No     22.557989
Yes    22.215190
Name: DistanceFromHome (KM), dtype: float64

In [17]:
data.columns

Index(['EmployeeID', 'FirstName', 'LastName', 'Gender', 'Age',
       'BusinessTravel', 'Department', 'DistanceFromHome (KM)', 'State',
       'Ethnicity', 'Education', 'EducationField', 'JobRole', 'MaritalStatus',
       'Salary', 'StockOptionLevel', 'OverTime', 'HireDate', 'Attrition',
       'YearsAtCompany', 'YearsInMostRecentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')